### PR TITLE
Test compute shader variables are zero initialized

### PR DESCRIPTION
**Pull request by austinEng**
<hr>

**Author checklist for test code/plans:**

- [x] All outstanding work is tracked with "TODO" in a test/file description or `.unimplemented()` on a test.
- [x] New helpers, if any, are documented using `/** doc comments */` and can be found via `helper_index.txt`.
- [ ] (Optional, sometimes not possible.) Tests pass (or partially pass without unexpected issues) in an implementation. (Add any extra details above.)

**[Reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md) for test code/plans:** (Note: feel free to pull in other reviewers at any time for any reason.)

- [x] The test path is reasonable, the [description](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) "describes the test, succinctly, but in enough detail that a reader can read only the test plans in a file or directory and evaluate the completeness of the test coverage."
- [x] Tests appear to cover this area completely, except for outstanding TODOs. Validation tests use control cases.
    (This is critical for coverage. Assume anything without a TODO will be forgotten about forever.)
- [x] Existing (or new) test helpers are used where they would reduce complexity.
- [x] TypeScript code is readable and understandable (is unobtrusive, has reasonable type-safety/verbosity/dynamicity).

### CODE REVIEW

**Comment by austinEng** [comment]
Currently produces hangs right now on both Windows / Metal.
Need to triage a bit more and may iterate to make suppressing those cases easier.
ex.) like make it possible to separate out structs w/ mixed types

**Comment by austinEng** [comment]
Ok @kainino0x this should be good to review. They all pass on D3D12. Non-workgroup tests pass on Mac Intel. Mac AMD has a problem (crbug.com/tint/1094). and overall on Metal, workgroup memory types are missing constructors for matrices and vectors; so most of them fail

File: src/webgpu/shader/execution/zero_init.spec.ts [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+                if (!isAtomic) {
+                  // Vector types
+                  for (const vectorType of kVectorContainerTypes) {
+                    // Fewer subcases: For nested types, only include
+                    // vec2<u32>, vec3<i32>, and vec4<f32>
+                    if (p._containerDepth > 0) {
+                      if (
+                        (vectorType !== 'vec2' || scalarType !== 'u32') &&
+                        (vectorType !== 'vec3' || scalarType !== 'i32') &&
+                        (vectorType !== 'vec4' || scalarType !== 'f32')
```
Comment (kainino0x) [inline src/webgpu/shader/execution/zero_init.spec.ts:132]: nit: imo clearer as `if (!((_ && _) || (_ && _) || (_ && _)))`

**Review by kainino0x — APPROVED** [review]

File: src/webgpu/shader/execution/zero_init.spec.ts (hunk 1 above)
Comment (austinEng) [reply src/webgpu/shader/execution/zero_init.spec.ts]: Done.

**Pull request merged**